src/app/modules/category/components/category-item/category-item.component.ts

```diff
@@ -0,0 +1,159 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    OnDestroy,
+    OnInit,
+} from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { BehaviorSubject, combineLatest, Subject } from 'rxjs';
+import { map, take, takeUntil } from 'rxjs/operators';
+import * as CategoryConstants from '../../category.constants';
+import * as CategoryActions from '../../state/actions/category.actions';
+import { CategoryProductData } from '../../state/category.models';
+import * as CategorySelectors from '../../state/selectors/category.selectors';
+
+@Component({
+    selector: 'app-category-item',
+    templateUrl: './category-item.component.html',
+    styleUrls: ['./category-item.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CategoryItemComponent implements OnInit, OnDestroy {
+    private readonly unsubscribe$ = new Subject();
+
+    filterProducts = [
+        {
+            value: CategoryConstants.popularValue,
+            viewValue: 'CATEGORY.POPULAR',
+        },
+        {
+            value: CategoryConstants.cheapValue,
+            viewValue: 'CATEGORY.CHEAP',
+        },
+        {
+            value: CategoryConstants.expensiveValue,
+            viewValue: 'CATEGORY.EXPENSIVE',
+        },
+        {
+            value: CategoryConstants.releaseDateValue,
+            viewValue: 'CATEGORY.NEW',
+        },
+        {
+            value: CategoryConstants.reviewsValue,
+            viewValue: 'CATEGORY.WITH_REVIEWS',
+        },
+    ];
+
+    selected = CategoryConstants.popularValue;
+
+    readonly categoryProducts$ = this.store
+        .select(CategorySelectors.selectCategoryProducts)
+        .pipe(
+            map(data => data),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly categoryProductTitle$ = this.store
+        .select(CategorySelectors.selectCategoryTitle)
+        .pipe(
+            map(data => data),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly filteredDataSource$ = new BehaviorSubject<
+        Array<CategoryProductData>
+    >(null);
+
+    readonly filteredData$ = this.filteredDataSource$
+        .asObservable()
+        .pipe(takeUntil(this.unsubscribe$));
```

Comment:
takeUntill should not be applied for the observables created from behaviorSubject, instead of that you should complete bwhaviorSubject in ngOnDestroy